### PR TITLE
Upgrade to libphonenumber-for-php v8.0 and fix phpunit warnings

### PR DESCRIPTION
**Pull request by anewmanjones**
This upgrades libphonenumber-for-php to the latest version and I also noticed some fixes to the test cases were needed.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
         "php": ">=5.4.0",
         "illuminate/support": "^4.0|^5.0",
         "illuminate/validation": "^4.0|^5.0",
-        "giggsey/libphonenumber-for-php": "^7.0",
+        "giggsey/libphonenumber-for-php": "^8.0",
```
Comment (Propaganistas) [inline composer.json]: Please change this to `^7.0|^8.0`, as v7 will still work of course.

Comment (Propaganistas) [inline composer.json]: Please change this to `^7.0|^8.0`, as v7 will still work of course.

File: tests/PhoneValidatorTest.php [hunk 2]
```diff
@@ -11,14 +11,14 @@ class PhoneValidatorTest extends TestCase
 
     protected $validator;
 
-    protected function getPackageProviders()
+    protected function getPackageProviders($app)
```
Comment (Propaganistas) [inline tests/PhoneValidatorTest.php]: Holding off on this. Omitting the `$app` parameter will generate a non-blocking warning for Laravel 5 builds, but allows to use the same tests file for Laravel 4.

File: tests/PhoneValidatorTest.php [hunk 3]
```diff
@@ -11,14 +11,14 @@ class PhoneValidatorTest extends TestCase
...
     {
         return [
             'Propaganistas\LaravelPhone\LaravelPhoneServiceProvider',
         ];
     }
 
-    protected function getPackageAliases()
+    protected function getPackageAliases($app)
```
Comment (Propaganistas) [inline tests/PhoneValidatorTest.php]: See previous comment

File: tests/PhoneValidatorTest.php [hunk 4]
```diff
@@ -345,10 +346,11 @@ public function testValidatePhoneLenient()
         ]));
     }
 
+    /**
+     * @expectedException \Propaganistas\LaravelPhone\Exceptions\InvalidParameterException
+     */
     public function testValidatePhoneFaultyParameters()
     {
-        $this->setExpectedException('Propaganistas\LaravelPhone\Exceptions\InvalidParameterException');
```
Comment (Propaganistas) [inline tests/PhoneValidatorTest.php]: Please change into this the `setExpectedException()` method and update composer.json accordingly to require phpunit `^5.2`

Comment (anewmanjones) [inline tests/PhoneValidatorTest.php]: The reason I removed this was because it has been deprecated in phpunit 5.2. However I will add this back shortly.

Comment (Propaganistas) [inline tests/PhoneValidatorTest.php]: Oops sorry, bold typo, I actually meant the `expectException()` method which is present since 5.2.

**Comment by Propaganistas** [comment]
Thanks. I've rolled out a new release.

In the end I reverted back to `setExpectedException()` since `expectException()` is only available since phpunit 5.2, but 5.2 requires >= PHP 5.6. And as I'm still supporting Laravel 4 (which allows PHP 5.5), I'd like to keep a single tests configuration. So we'll deal with the deprecation once it's effectively being removed from phpunit :)

**Pull request merged**